Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -241,8 +242,14 @@ void CookieStore::set(CookieInit&& options, Ref<DeferredPromise>&& promise)
             promise->reject(Exception { TypeError, "The path must begin with a '/'"_s });
             return;
         }
+
         if (!cookie.path.endsWith('/'))
             cookie.path = cookie.path + '/';
+
+        if (cookie.path.utf8().length() > maximumAttributeValueSize) {
```

Comment:
It would be nice if we didn't need to allocate and encode just to get the encoded length.  I started this in https://bugs.webkit.org/show_bug.cgi?id=232962 .  That's out of scope in this PR, but might be worth adding a comment to this and the domain.utf8()

Reply:
Also, add a test for this.

Reply:
And that test should be added to the WPT repo.

Reply:
Ok I've added a comment about the possible encoded length improvement. I've also added testing for the path and domain lengths. Ideally, this would be part of the `cookieStore_set_arguments.https.any.html` test since that's where we test all of the ways of setting a cookie with the different arguments (like path and domain). This is where I've added it for the moment. But I'm not sure if we're allowed to just modify the existing tests like that or how I can add this to WPT?

Reply:
Yeah, just modify the test copies in our repo like you did, then make another PR to https://github.com/web-platform-tests/wpt